apps/prairielearn/src/lib/regrading.js

```diff
@@ -6,100 +6,73 @@ const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'
-    };
-    createJobSequence(options, function (err, job_sequence_id) {
-      if (ERR(err, callback)) return;
-      callback(null, job_sequence_id);
+  var options = {
+    courseId: course_id,
+    courseInstanceId: course_instance_id,
+    assessmentId: assessment_id,
+    userId: user_id,
+    authnUserId: authn_user_id,
+    type: 'regrade_assessment_instance',
+    description: 'Regrade ' + assessment_instance_label + ' for ' + jobInfo,
+  };
+  const serverJob = await createServerJob(options);
 
-      // We've now triggered the callback to our caller, but we
-      // continue executing below to launch the jobs themselves.
+  // We've now triggered the callback to our caller, but we
+  // continue executing below to launch the jobs themselves.
 
-      var jobOptions = {
-        course_id: course_id,
-        course_instance_id: course_instance_id,
-        assessment_id: assessment_id,
-        user_id: user_id,
-        authn_user_id: authn_user_id,
-        type: 'regrade_assessment_instance',
-        description: 'Regrade ' + assessment_instance_label + ' for ' + jobInfo,
-        job_sequence_id: job_sequence_id,
-        last_in_sequence: true,
-      };
-      createJob(jobOptions, function (err, job) {
-        if (err) {
-          logger.error('Error in createJob()', err);
-          failJobSequence(job_sequence_id);
-          return;
-        }
-        job.verbose('Regrading ' + assessment_instance_label + ' for ' + jobInfo);
-        var params = [assessment_instance_id, authn_user_id];
-        sqldb.call('assessment_instances_regrade', params, function (err, result) {
-          if (ERR(err, function () {})) {
-            job.fail(err);
-          } else {
-            if (result.rowCount !== 1) {
-              job.fail(new Error('Incorrect rowCount: ' + result.rowCount));
-            }
-            job.verbose('Regrading complete');
-            var regrade = result.rows[0];
-            if (regrade.updated) {
-              job.verbose('Questions updated: ' + regrade.updated_question_names.join(', '));
-              job.verbose(
-                'New score: ' +
-                  Math.floor(regrade.new_score_perc) +
-                  '% (was ' +
-                  Math.floor(regrade.old_score_perc) +
-                  '%)',
-              );
-            } else {
-              job.verbose('No changes made');
-            }
-            ltiOutcomes.updateScore(assessment_instance_id, (err) => {
-              if (err) {
-                job.fail(err);
-              } else {
-                job.succeed();
-              }
-            });
-          }
-        });
-      });
-    });
+  await serverJob.executeInBackground(async (job) => {
+    job.verbose('Regrading ' + assessment_instance_label + ' for ' + jobInfo);
+    var params = [assessment_instance_id, authn_user_id];
+    const jobResult = await sqldb.callAsync('assessment_instances_regrade', params);
```

Comment:
Nit: use `queryRow` with Zod validation for bonus points 🙂

Reply:
Updated to use Zod validation in commit [ecafed4](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/ecafed4ec68b0006158dd7f2866a9331734aa370). Used `callRow` here instead of `queryRow` as this is for a sproc.